src/services/services.ts

```diff
@@ -1798,7 +1798,7 @@ namespace ts {
             const span = createTextSpanFromBounds(start, end);
             const formatContext = formatting.getFormatContext(formatOptions);
 
-            return flatMap(deduplicate(errorCodes, equateValues, compareValues), errorCode => {
+            return flatMap(deduplicate<number>(errorCodes, equateValues, compareValues), errorCode => {

```

Comment:
Here's what's happening: Previously we'd infer `number` from `errorCodes`, then fix that inference as we contextually typed `equateValues`, and finally _ignore_ inferences from `compareValues` because of fixing. Now we defer processing of `equateValues` to get the best possible contextual type, which means we'll infer from both `errorCodes` and `compareValues`. And from `compareValues` we get the candidate `number | undefined` which is technically contravariant, but we ignore that because we're not in `--strictFunctionTypes` mode. So, we end up with a worse inference. The real fix here would be for us to do the work to use `--strictFunctionTypes` in the compiler, the legacy rules just aren't right but it is a breaking change (in other places) to fix them.

Reply:
It's probably too far gone to change; but is there a good compat reason we can't perform inferences the same way outside and inside `strictFunctionTypes` mode?

But aight, if that's the case I guess it can't be avoided. :(